Sources/ConsoleKit/Utilities/GenerateAutocompleteCommand.swift

```diff
@@ -0,0 +1,61 @@
+import Foundation
+
+struct GenerateAutocompleteCommand: Command {
+
+    var help: String { "Generate shell completion scripts for the executable" }
+
+    var rootCommand: AnyCommand?
+
+    init(rootCommand: AnyCommand? = nil) {
+        self.rootCommand = rootCommand
+    }
+
+    struct Signature: CommandSignature {
+
+        @Option(
+            name: "shell",
+            short: "s",
+            help: "Generate completion script for SHELL [ \(Shell.allCases.map { "\($0)" }.joined(separator: " | ")) ]. \n"
+                + "Defaults to the \"SHELL\" environment variable if possible."
+            ,
+            completion: .values(of: Shell.self)
+        )
+        var shell: Shell?
+
+        @Option(
+            name: "output",
+            short: "o",
+            help: "Write output to file at OUTPUT. \n"
+                + "Defaults to printing to stdout.",
+            completion: .files()
+        )
+        var output: String?
+    }
+
+    func run(using context: CommandContext, signature: Signature) throws {
+        guard let rootCommand = self.rootCommand else { fatalError("`rootCommand` was not initialized") }
+        guard let shell = signature.shell ?? self.environmentShell() else {
+            throw CommandError.missingRequiredArgument(signature.$shell.name)
```

Comment:
Is there a `.missingRequiredOption` error that could be used instead?

Reply:
No, as far as I can tell, a required `Option` is not currently supported by ConsoleKit. Would be a great enhancement though!